File: LeetCode/Easy/0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i = 0;
        double sum = 0;

        // 첫번째 평균값
        for(; i < k; ++i) {
            sum += nums[i];
        }

        double maxSum = sum;

        while(i < nums.length) {

            sum += (nums[i] - nums[i - k]);

            maxSum = Math.max(maxSum, sum);

            ++i;
        }

        return maxSum / k;
    }
}

File: LeetCode/Easy/0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.kt
class Solution {
    fun findMaxAverage(nums: IntArray, k: Int): Double {
        var i = 0
        var sum = 0.0

        for(n in 0 .. k - 1) {
            sum += nums[n]
            ++i
        }

        var maxSum = sum

        while(i < nums.size) {

            sum += nums[i] - nums[i - k]

            maxSum = Math.max(maxSum, sum)

            ++i

        }

        return maxSum / k
    }
}